test/show.jl

```diff
@@ -1,28 +1,37 @@
 @testset "Show" begin
     Mod.@polyvar x y z
-    @test sprint(show, (x*y^2 + x + 1 + y)) == "xy^2 + x + y + 1"
-    @test sprint(show, (x + 1 + y) / x^2) == "(x + y + 1) / (x^2)"
-    @test sprint(show, (x - y - x + y) / (x^2 - x)) == "(0) / (x^2 - x)"
+    @test sprint(show, (x*y^2 + x + 1 + y)) == "xy² + x + y + 1"
+    @test sprint(show, (x + 1 + y) / x^2) == "(x + y + 1) / (x²)"
+    @test sprint(show, (x - y - x + y) / (x^2 - x)) == "(0) / (x² - x)"
     # Test taken from TypedPolynomials
     @test sprint(show, x) == "x"
     @test sprint(show, x^0) == "1"
     @test sprint(show, x^1) == "x"
-    @test sprint(show, x^2) == "x^2"
-    @test sprint(show, 1x^2) == "x^2"
+    @test sprint(show, x^2) == "x²"
+    @test sprint(show, 1x^2) == "x²"
     @test sprint(show, 5x) == "5x"
     @test sprint(show, x * y) == "xy"
     @test sprint(show, y * x) == "xy"
     @test sprint(show, 5 + y + x) == "x + y + 5"
     @test sprint(show, y + 5 + x) == "x + y + 5"
-    @test sprint(show, x + x^2) == "x^2 + x"
-    @test sprint(show, x^2 + x) == "x^2 + x"
-    @test sprint(show, x^2 - 3.0x) == "x^2 - 3.0x"
-    @test sprint(show, -x^3) == "-x^3"
-    @test sprint(show, -x^3 + y^2 - x) == "-x^3 + y^2 - x"
-    @test sprint(show, -2.0x^2) == "-2.0x^2"
+    @test sprint(show, x + x^2) == "x² + x"
+    @test sprint(show, x^2 + x) == "x² + x"
+    @test sprint(show, x^2 - 3.0x) == "x² - 3.0x"
+    @test sprint(show, -x^3) == "-x³"
+    @test sprint(show, -x^3 + y^2 - x) == "-x³ + y² - x"
+    @test sprint(show, -2.0x^2) == "-2.0x²"
     @test sprint(show, (1.0 + 3.1im) * x*z) == "(1.0 + 3.1im)xz"
     @test sprint(show, -(1.0 + 3.1im) * z*x) == "(-1.0 - 3.1im)xz"
-    @test sprint(show, x^2 + (1.0 + 3.1im) * x) == "x^2 + (1.0 + 3.1im)x"
-    @test sprint(show, x^2 - (1.0 + 3.1im) * x) == "x^2 + (-1.0 - 3.1im)x"
+    @test sprint(show, x^2 + (1.0 + 3.1im) * x) == "x² + (1.0 + 3.1im)x"
+    @test sprint(show, x^2 - (1.0 + 3.1im) * x) == "x² + (-1.0 - 3.1im)x"
     @test sprint(show, [1.0, 2.0] * x) == "([1.0, 2.0])x"
+
+    # This is only supported by DynamicPolynomials so far
+    # @polyvar x[0:9]
+    # @test sprint(show, sum(i*x[i]^i for i=1:10)) == "10x₉¹⁰ + 9x₈⁹ + 8x₇⁸ + 7x₆⁷ + 6x₅⁶ + 5x₄⁵ + 4x₃⁴ + 3x₂³ + 2x₁² + x₀"
```

Comment:
These would fail with TypedPolynomials right now

Reply:
What does it do ?

Reply:
The things is `TypedPolynomials` currently would create variables with name `x1`, `x2`, ...
whereas the code only changes to unicode if the variable has name `x[i]` (resp. `x[i,j]` etc.).
Also note that TypedPolynomials currently only supports the vector case and not general variable tensors. So this would also cause trouble a couple of lines down.

Reply:
I assume the tuple behaviour also corresponds to some type stability stuff since puttings things in an array basically trashes this (since each variable is of a different type).

Reply:
A more general question is, how to handle tests when there is no feature parity.

Reply:
We could add a function in the api that returns the base name and a vector of indices

Reply:
That would be one possibility, but still so far TypedPolynomials doesn’t support higher dimensional variable arrays, so we would also need to add these

Reply:
No it would simply return a vector of only one element

Reply:
```julia
using TypedPolynomials
@polyvar x[1:2]
name_indices(x[1]) # -> [1]
@polyvar y
name_indices(y) # -> []
```